backend/services/event.py

```diff
@@ -189,3 +242,459 @@ def delete(self, subject: User, id: int) -> None:
+            attendee: User registered for the event
+            event: EventDetails of the event seeking registration for
+
+        Returns:
+            EventRegistration or None if no registration found
+
+        Raises:
+            UserPermissionException if subject does not have permission
+        """
+        # Feature-specific authorization: User is unregistering themself
+        # Administrative Permission: organization.events.manage : organization/{id}
+        if subject.id != attendee.id:
+            self._permission.enforce(
+                subject,
+                "organization.events.*",
+                f"organization/{event.organization.id}",
+            )
+
+        # Query for EventRegistration
+        event_registration_entity = (
+            self._session.query(EventRegistrationEntity)
+            .where(EventRegistrationEntity.user_id == attendee.id)
+            .where(EventRegistrationEntity.event_id == event.id)
+            .one_or_none()
+        )
+
+        # Return EventRegistration model or None
+        if event_registration_entity is not None:
+            return event_registration_entity.to_model()
+        else:
+            return None
+
+    def is_user_an_organizer(self, user: User, event: EventDetails) -> bool:
+        """
+        Test whether a user is an organizer of an event.
+
+        Args:
+            user: The user to check on event organizer status of.
+            event: The event in question.
+
+        Returns:
+            bool True if user is an organizer, False otherwise.
+        """
+        registration = self.get_registration(user, user, event)
+        return registration.is_organizer if registration else False
+
+    def get_registrations_of_event(
+        self, subject: User, event: EventDetails
+    ) -> list[EventRegistration]:
+        """
+        List the registrations of an event.
+
+        This API endpoint currently requires the subject to be registered as the
+        organizer of an event or have administrative permission of action
+        "organization.events.manage" for "organization/{organization id}".
+
+        Args:
+            subject: The authenticated user making the request.
+            event: The event whose registrations are being queried.
+
+        Returns:
+            list[EventRegistration]
+
+        Raises:
+            UserPermissionException if user is not an event organizer or admin.
+        """
+        if not self.is_user_an_organizer(subject, event):
+            self._permission.enforce(
+                subject,
+                "organization.events.*",
+                f"organization/{event.organization.id}",
+            )
+
+        event_registration_entities = (
+            self._session.query(EventRegistrationEntity)
+            .where(EventRegistrationEntity.event_id == event.id)
+            .all()
+        )
+
+        return [entity.to_model() for entity in event_registration_entities]
+
+    def set_event_organizer(

```

Comment:
Is there a way of controlling this from the front-end? Something we should add a ticket for, if not! Would be great to be able to elevate someone from the registrations list (e.g. for making a workshop lead an organizer).